selectors/views/channel_sidebar.ts

```diff
@@ -47,6 +47,18 @@ export const getCategoriesForCurrentTeam: (state: GlobalState) => ChannelCategor
     });
 })();
 
+export const getAutoSortedCategoryIds: (state: GlobalState) => string[] = (() => {
+    const getCategoriesForTeam = makeGetCategoriesForTeam();
+
+    return memoizeResult((state: GlobalState) => {
```

Comment:
`createSelector` would be better here since it will prevent recalculating when state changes but `categories` does not. The reason we didn't do that in some of the other selectors is because they didn't have any equivalent like to the filter or because they required chaining results of more.

```javascript
createSelector(
    (state) => getCategoriesForTeam(state, getCurrentTeamId(state)),
    (categories) => categories.filter(...),
);
```